bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/discovery/BridgeMDNSDiscoveryParticipant.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.zwavejs.internal.discovery;
+
+import static org.openhab.binding.zwavejs.internal.BindingConstants.*;
+
+import java.util.Objects;
+import java.util.Set;
+
+import javax.jmdns.ServiceInfo;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.mdns.MDNSDiscoveryParticipant;
+import org.openhab.core.i18n.TranslationProvider;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link zwavejs} is responsible for discovering new and removed Zwave JS servers. It uses the
+ * central {@link org.openhab.core.config.discovery.mdns.internal.MDNSDiscoveryService}.
+ *
+ * @author Leo Siepel - Initial contribution
+ */
+@Component(configurationPid = "discovery.zwavejs")
+@NonNullByDefault
+public class BridgeMDNSDiscoveryParticipant implements MDNSDiscoveryParticipant {
+
+    private static final String SERVICE_TYPE = "_zwave-js-server._tcp.local.";
+    private static final String MDNS_PROPERTY_HOME_ID = "homeId";
+
+    private final TranslationProvider translationProvider;
+
+    @Activate
+    public BridgeMDNSDiscoveryParticipant(@Reference TranslationProvider translationProvider) {
+        this.translationProvider = translationProvider;
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Set.of(THING_TYPE_GATEWAY);
+    }
+
+    @Override
+    public String getServiceType() {
+        return SERVICE_TYPE;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(ServiceInfo service) {
+        ThingUID uid = getThingUID(service);
+
+        if (Objects.nonNull(uid)) {
+            String host = service.getHostAddresses()[0];
+            String homeId = service.getPropertyString(MDNS_PROPERTY_HOME_ID);
+            String thingID = uid.getAsString().split(ThingUID.SEPARATOR)[1];
+            String label = translationProvider.getText(FrameworkUtil.getBundle(getClass()),
+                    "discovery.%s.label".formatted(thingID), null, null, host);
```

Comment:
No need to provide locale retrieved from LocaleProvider ?

Reply:
Got this from some other bindings and assumed that the translationprovider by default uses the set locale.
Was that wrong?

Reply:
Will give it a try this evening

Reply:
That is a good news, I just don't understand how it works to lead to this result. Will have to investigate...